### PR TITLE
added country code filter to export

### LINKED ISSUE #281: country extract
I would like to provide and build a smaller extract to use photon for testing purposes too. To achieve this we can filter the export by country, per @lonvia this should be as simple as adding
`WHERE country_code in ('de', 'ch', 'at')` in the two SQL queries in readEntireDatabase.
  
https://github.com/komoot/photon/commit/12b3f3f913203166785e581dba8f2aba78e7cdb9

### PR DESCRIPTION
**Pull request by karussell**
Fixes #281 

Will done an export for Switzerland, if ready, get it via 

```
wget -O - http://download1.graphhopper.com/public/photon-ch-180109.tar.bz2 | bzip2 -cd | tar x
```

I'll also do an export for France and Germany so that we can easier test it using the geocoder testing framework.

### CODE REVIEW

**Comment by karussell** [comment]
Hmmh, it does not look like the filter is active as the process is now doing its work for over 1 hour.

File: src/main/java/de/komoot/photon/nominatim/NominatimConnector.java [hunk 1]
```diff
@@ -353,7 +357,8 @@ public void processRow(ResultSet rs) throws SQLException {
 		});
 
 
-		template.query("SELECT place_id, osm_id, parent_place_id, startnumber, endnumber, interpolationtype, postcode, country_code, linegeo FROM location_property_osmline ORDER BY geometry_sector; ", new RowCallbackHandler() {
+		template.query("SELECT place_id, osm_id, parent_place_id, startnumber, endnumber, interpolationtype, postcode, country_code, linegeo FROM location_property_osmline " +
+				countryCodeStr + " ORDER BY geometry_sector; ", new RowCallbackHandler() {
```
Comment (lonvia) [inline src/main/java/de/komoot/photon/nominatim/NominatimConnector.java]: This looks wrong. It should be 'WHERE country_code ...' not 'AND country_code ...'.

File: src/main/java/de/komoot/photon/nominatim/NominatimConnector.java [hunk 2]
```diff
@@ -321,8 +321,12 @@ public void readEntireDatabase() {
 		final BlockingQueue<PhotonDoc> documents = new LinkedBlockingDeque<PhotonDoc>(20);
 		Thread importThread = new Thread(new ImportThread(documents));
 		importThread.start();
+		String countryCodeStr = "";
+		if(countryCodes.length > 0)
+			countryCodeStr = " AND country_code in ('" + String.join("','", countryCodes) +"')";
```
Comment (lonvia) [inline src/main/java/de/komoot/photon/nominatim/NominatimConnector.java]: I would lower-case the strings as Nominatim expects them this way. I would also suggest to do a bit of sanity checking of the parts: length must be exactly two characters and characters must be ascii letters only.

**Comment by karussell** [comment]
Thanks - will try your suggestion!

**Comment by karussell** [comment]
Looks like this now worked (~3h for Germany)
```
wget -O - http://download1.graphhopper.com/public/photon-de-180109.tar.bz2 | bzip2 -cd | tar x
```

**Comment by karussell** [comment]
Have tested this extract locally with success!

**Pull request merged**